src/zarr/registry.py

```diff
@@ -131,6 +182,16 @@ def register_buffer(cls: type[Buffer]) -> None:
     __buffer_registry.register(cls)
 
 
+def register_v3dtype(cls: type[ZarrDType]) -> None:
+    assert cls.zarr_spec_format == "3"
+    __v3_dtype_registry.register(cls, class_registry_key=cls._zarr_spec_identifier)
+
+
+def register_v2dtype(cls: type[ZarrDType]) -> None:
```

Comment:
Ideally, we could also backport this to Zarr V2? I'll admit that I haven't really researched this much though. Putting it here just to start a discussion.

Reply:
I don't think there is a strong desire in the community to further evolve the v2 spec.